### PR TITLE
Parameterized Rules

### PR DESCRIPTION
**Pull request by jskrajareddy21**
## Description

This PR addresses issue: https://github.com/Nike-Inc/spark-expectations/issues/51

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/core/test_expectations.py [hunk 1]
```diff
@@ -2062,7 +2062,81 @@ def _error_threshold_exceeds(expectations):
                                      {"row_dq_status": "Passed", "source_agg_dq_status": "Passed",
                                       "final_agg_dq_status": "Passed", "run_status": "Passed",
                                       "source_query_dq_status": "Passed", "final_query_dq_status": "Passed"}  # status
-                             )
+                             ),
+                            (
+                             # Test case 23
```
Comment (asingamaneni) [inline tests/core/test_expectations.py:2067]: Can you please description about `parameterized query` example in the description, so that it is easily recognized - while we read this later

File: tests/core/test_expectations.py [hunk 2]
```diff
@@ -2105,7 +2179,7 @@ def test_with_expectations(input_df,
     # Decorate the mock function with required args
     @se.with_expectations(
         "dq_spark.test_final_table",
-        user_conf={user_config.se_notifications_on_fail: False},
+        user_conf={user_config.se_notifications_on_fail: False, user_config.se_dq_rules_params: {'table' : 'test_table'} },
```
Comment (asingamaneni) [inline tests/core/test_expectations.py:2183]: Can we also test a negative scenario which should raise exception, when you don't pass user_config in this use_case!

File: tests/core/test_context.py [hunk 3]
```diff
@@ -1655,3 +1656,13 @@ def test_set_enable_error_table():
...
     context.set_se_enable_error_table(False)
     assert context.get_se_enable_error_table is False
+
+
+def test_set_dq_rules_params():
+    # default case is True for enabling error table
+    context = SparkExpectationsContext(product_id="product1", spark=spark)
+    assert context.get_dq_rules_params == {}
+
+    # testing for False do not write error records in error table
```
Comment (asingamaneni) [inline tests/core/test_context.py]: Please update the comments to appropriately reflect the test scenario

File: spark_expectations/utils/reader.py [hunk 4]
```diff
@@ -149,6 +150,7 @@ def get_rules_from_df(
             target_table: Provide the full table name for which the data quality rules are being run
             is_dlt: True if this for fetching the rules for dlt job
             tag: If is_dlt is True, provide the KPI for which you are running the data quality rule
+            params:
```
Comment (asingamaneni) [inline spark_expectations/utils/reader.py]: Please update the docustring

**Review by asingamaneni — CHANGES_REQUESTED** [review]
Please update the project documentation to reflect the latest changes and functionality

File: tests/core/test_expectations.py (hunk 1 above)
Comment (jskrajareddy21) [reply tests/core/test_expectations.py:2067]: added comments in description

File: tests/core/test_context.py (hunk 3 above)
Comment (jskrajareddy21) [reply tests/core/test_context.py]: updated comments properly

File: spark_expectations/utils/reader.py (hunk 4 above)
Comment (jskrajareddy21) [reply spark_expectations/utils/reader.py]: updated docstring

File: tests/core/test_expectations.py (hunk 2 above)
Comment (jskrajareddy21) [reply tests/core/test_expectations.py:2183]: added negative test scenario

**Review by asingamaneni — APPROVED** [review]
LGTM
